pytorch_lightning/trainer/trainer.py

```diff
@@ -451,7 +451,7 @@ def __init__(
             amp_level,
             plugins,
         )
-        self.logger_connector = LoggerConnector(self, log_gpu_memory)
+        self.logger_connector: LoggerConnector = LoggerConnector(self, log_gpu_memory)
```

Comment:
nit: not required

```suggestion
        self.logger_connector = LoggerConnector(self, log_gpu_memory)
```

Reply:
@awaelchli, did you add this to fix mypy?

Reply:
unfortunately, typing breaks if this is removed.

Reply:
I just pulled the PR, removed the annotation, and ran mypy (version 0.910). Everything is green

```suggestion
        self.logger_connector = LoggerConnector(self, log_gpu_memory)
```

Reply:
Awesome! I saw you have a mypy PR out, should I rebase on top of that after it merges and remove this?

Reply:
You can remove this already. This PR just needs to rebase on top of #11095